sdks/python/apache_beam/typehints/arrow_type_compatibility.py

```diff
@@ -348,3 +348,37 @@ def _from_serialized_schema(serialized_schema):
   def __reduce__(self):
     return self._from_serialized_schema, (
         self._beam_schema.SerializeToString(), )
+
+
+class PyarrowArrayBatchConverter(BatchConverter):
+  def __init__(self, element_type: type):
+    super().__init__(pa.Array, element_type)
+    self._element_type = element_type
+    beam_fieldtype = typing_to_runner_api(element_type)
+    self._arrow_type = _arrow_type_from_beam_fieldtype(beam_fieldtype)
+
+  @staticmethod
+  @BatchConverter.register
+  def from_typehints(element_type,
+                     batch_type) -> Optional['PyarrowArrayBatchConverter']:
+    if batch_type == pa.Array:
+      return PyarrowArrayBatchConverter(element_type)
+
+    return None
+
+  def produce_batch(self, elements):
+    return pa.array(list(elements), type=self._arrow_type)
```

Comment:
Out of curiosity, if the user wants to add other flags to the pa.array, such as `size` or `mask` which are optional for pa.array, how would they do it?

Reply:
Good question. Unfortunately they wouldn't be able to do that in this framework.

They could work around it though by making a DoFn that yields batches and constructing the `pa.Array` themselves.

Note we use this code internally whenever we need to implicitly create a batch for the user.